### ISSUE TITLE
[BUG] Wireguard fails to start on a clean install of Raspbian

### ISSUE THREAD

**Issue by orazioedoardo** [pivpn/pivpn#924]
## In raising this issue, I confirm the following: 

`{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting can be *is* directly related to the pivpn installer script.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

**Describe the bug**
Wireguard does not start, I get this error at the end of the installation:
```
::: Install Complete...
::: Restarting services...
Job for wg-quick@wg0.service failed because the control process exited with error code.
See "systemctl status wg-quick@wg0.service" and "journalctl -xe" for details.
:::    Checking for unattended-upgrades... not installed!
:::    Package unattended-upgrades successfully installed!
:::
::: Installing scripts to /opt/pivpn...
 done.
::: Flushing writes to disk...
::: done.
:::
```

**To Reproduce**
1. Flash Raspbian
2. Install PiVPN and choose WireGuard
3. See error

**Expected behavior**
The WireGuard unit should run.

**Additional context**
This is a clean installation of Raspbian, with no additional software.

**Comment by orazioedoardo** [comment]
The problem is that DKMS compiles the wireguard module using the kernel headers from the latest version of the kernel:
```
$ dkms status
wireguard, 0.0.20200121, 4.19.93+, armv7l: installed
wireguard, 0.0.20200121, 4.19.93-v7+, armv7l: installed
wireguard, 0.0.20200121, 4.19.93-v7l+, armv7l: installed
```
However, the latest Raspbian image `2019-09-26-raspbian-buster-lite.img` ships with kernel `4.19.75-v7`.

Solution, upgrade all packages and reboot (upgrading just `raspberrypi-kernel` might be enough):
```
sudo apt update
sudo apt upgrade
sudo reboot
```

**Comment by orazioedoardo** [comment]
To solve, first upgrade all packages with: `sudo apt update && sudo apt upgrade -y`, then reboot with `sudo reboot` and follow these steps: https://github.com/pivpn/pivpn/issues/920#issuecomment-578651638.

You won't run into this issue if you upgrade the system and reboot BEFORE running the PiVPN install script.

**Comment by orazioedoardo** [comment]
Now telling the user to upgrade all packages before proceeding: https://github.com/pivpn/pivpn/commit/f2fa01e3a518e1722f7b90e0e90e04aa2027d5c3

**Comment by 4s3ti** [comment]
I guess there is no other way to do it @orazioedoardo even tough we update packages during the install, if kernel is there to be upgraded the reboot is always needed .. we never felt this need and neither did pihole because we never had anything working directly on the kernel so far which is the case with Wireguard .. so this issue is expected to happen ... i guess your solution is the correct one =)

**Comment by orazioedoardo** [comment]
What I empirically found is that you don’t need to boot into the new kernel to successfully build the module, you just need to have, on disk, matching kernel and kernel headers (I don’t really know what is the underlying reason though).

So I tried to upgrade the kernel and then immediately install wireguard-dkms. It did work however iptables was complaining about unknown ‘nat’ table and ‘comment’ module or something.

I wanted to have so that you only need to reboot at the end of the script but it doesn’t seem possible.

I also wanted to know, at minimum, what packages need to be upgraded (maybe only the kernel) but then I got lazy and just decided to upgrade all packages. In the end, if I have to reboot, I’ll take the opportunity and upgrade everything.
